### ISSUE TITLE
Feature Request: Validator client logs should scrub beacon endpoint data

### ISSUE THREAD

**Issue by djoyahoy** [sigp/lighthouse#2276]
## Description

The validator client logs the entire beacon http endpoint including basic auth fields. It should scrub sensitive information when logging.

## Version

Lighthouse v1.2.1-b34a79d
rustc 1.50.0

## Present Behaviour

Log lines which print the current beacon endpoint include auth data. For example:
```
INFO Connected to beacon node                endpoint: https://...
ERRO Failed to download proposer duties      err: All endpoints failed https://...
WARN Beacon node is not synced               endpoint: https://...
etc.
```

## Expected Behaviour

The logs should scrub that information and simply output the URL without basic auth included.

## Steps to resolve

N/A

**Comment by djoyahoy** [comment]
This behavior also appears in some of the prometheus metrics output by the `/metrics` endpoints. For example, the metrics `eth1_endpoint_requests` and `eth1_endpoint_errors`.

**Comment by seh** [comment]
I confirmed that the traces published via OpenCensus **don't** include these details.

**Comment by paulhauner** [comment]
Perhaps we can just include the base URL (i.e., http://infura.io) instead of the entire path. It is useful to know which endpoint is doing what.

**Comment by macladson** [comment]
I'm going to jump on this.

**Comment by paulhauner** [comment]
Resolved in #2326, thanks @macladson.
